gap/PolygonalComplexes/modification.gi

```diff
@@ -1879,6 +1879,144 @@ InstallMethod( SplitAllVertices, "for a polygonal complex",
 );
 RedispatchOnCondition( SplitAllVertices, true, [IsTwistedPolygonalComplex], [IsPolygonalComplex], 0 );
 
+InstallMethod( ButterflyInsertionSurface,
+"for a simplicial surface and a pair of edges or a vertex list", [IsSimplicialSurface,IsList],
+
+     function( surface, t )
+
+    local bound, path1, path2, newSurface, butterfly,
+          edges, newverts, w, i, e, v, ime;
+
+    if Length(t) = 2 then
+        if  not IsSubset(Edges(surface),t) or 
+        Length(CommonVerticesOfEdgesNC(surface,t[1],t[2])) = 0 then
+          ErrorNoReturn("Requires a  pair of adjacent edges");
+        fi;
+       t := Union(VerticesOfEdge(surface,t[1]),VerticesOfEdge(surface,t[2]));
+       if not t[2]  in CommonVerticesOfEdgesNC(surface,t[1],t[2]) then
+           w := t[2]; t[2] := t[1]; t[1] := w;
+       fi;
+    elif Length(t) = 3 then
+       # we expect a vertex path of length 3
+       if not IsSubset(Vertices(surface),t) or
+       not IsVerticesAdjacent(surface, t[1], t[2] ) or
+       not IsVerticesAdjacent(surface, t[2], t[3] ) then
+        ErrorNoReturn("Requires vertex path of length 3");fi;
+    fi;
+
+
+    # Define a butterfly
+    butterfly:=SimplicialSurfaceByDownwardIncidence(
+           [[1,2],[2,4],[3,4],[1,3],[2,3]],[[1,5,4],[2,3,5]]);
+
+    # find the path along which we split
+    path1 := VertexEdgePathByVertices(surface,t);
+    edges := EdgesAsList(path1);
+    # split along the middle vertex of the path
+    if IsInnerVertex(surface,t[2]) then
+        newSurface := SplitEdgePath(surface,path1);
+        # find the new path of the split surface
+        bound := newSurface[2];
+        # and the split surface
+        newSurface := newSurface[1];
+        # this is now the newly created path of length 4
+        path1:= ConcatenationOfPaths(newSurface,
+            bound[1][1],ReversedPath(bound[2][1]));
+        path2:=VertexEdgePathByEdges(butterfly,[1,2,3,4]);
+
+        return JoinVertexEdgePaths(newSurface,path1,butterfly,path2);
+	        
+    elif IsInnerEdge(surface,edges[1]) and IsInnerEdge(surface,edges[2]) then
+
+        newSurface := SplitVertexEdgePath(surface, path1);
+        bound := newSurface[2];
+        Assert(0, Length(bound) = 3,"incorrect path");
+        # and the split surface
+        newSurface := newSurface[1];
+        # the images of the split edges are
+	e := List( bound, x-> List(x,EdgesAsList));
+	v := List( bound, x-> List(x,VerticesAsList));
+	i := First([1..3] , x-> Length(e[x][1])=1 and edges[1] in e[x][2] );
+        Assert(0,i<>fail,"incorrect path");
```

Comment:
Shouldn't `Assert(0,i=fail,"incorrect path")` be here?

Reply:
I think i<>fail, as we want i not to be fail.